qubesadmin/tools/qvm_start_gui.py

```diff
@@ -177,19 +181,22 @@ def common_guid_args(self, vm):
 
     @staticmethod
     def guid_pidfile(xid):
-        '''Helper function to construct a pidfile path'''
+        """Helper function to construct a pidfile path"""
         return '/var/run/qubes/guid-running.{}'.format(xid)
 
     @asyncio.coroutine
     def start_gui_for_vm(self, vm, monitor_layout=None):
-        '''Start GUI daemon (qubes-guid) connected directly to a VM
+        """Start GUI daemon (qubes-guid) connected directly to a VM
 
         This function is a coroutine.
 
         :param vm: VM for which start GUI daemon
         :param monitor_layout: monitor layout to send; if None, fetch it from
             local X server.
-        '''
+        """
+        if vm.guivm != vm.app.get_local_name():
```

Comment:
I think this needs to be (and already is) handled by a function calling it. But looks harmless.